account_bill_line_distribution/models/account_invoice.py

```diff
@@ -0,0 +1,107 @@
+                        if distrib.company_id != self.company_id:
+                            # Debit line for the amount due from the other
+                            # company
+                            from_lines.append({
+                                'name': invoice_line_id.name,
+                                'debit': amount,
+                                'account_id':
+                                    inv.company_id.due_from_account_id.id,
+                                'partner_id':
+                                    distrib.company_id.partner_id.id})
+                            # Credit line for the current company
+                            from_lines.append({
+                                'name': invoice_line_id.name,
+                                'credit': amount,
+                                'account_id':
+                                    invoice_line_id.account_id.id,
+                                'partner_id':
+                                    inv.company_id.partner_id.id})
+
+                # Create Journal Entry in the current company
+                if from_lines:
+                    from_move = account_move.sudo().create(from_move_value)
+                    for line in from_lines:
+                        line.update({'move_id': from_move.id})
+                    self.env['account.move.line'].create(from_lines)
+                    from_move.post()
+
+                # Due To's
+                for company in companies:
+                    to_lines = []
+                    # Skip the current company because that has been taken
+                    # care of already
+                    if company.id != inv.company_id.id:
+                        to_move_vals = {
+                            'journal_id':
+                                company.due_fromto_payment_journal_id.id,
+                            'state': 'draft',
+                            'ref': inv.company_id.name + ': ' + inv.number,
+                            'company_id': company.id}
+                        # Credit company "Due_To" Account with debit
+                        # amount from its previous "Due from" Account entry
+                        for line in from_lines:
+                            if line['partner_id'] == company.partner_id.id:
+                                to_lines.append({
+                                    'name': line['name'],
+                                    'credit': line['debit'],
+                                    'account_id': company.due_to_account_id.id,
+                                    'partner_id':
+                                        inv.company_id.partner_id.id})
+                                # Debit Account of invoice line
+                                account_obj = self.env['account.account']
+                                code = \
+                                    account_obj.browse(line['account_id']).code
```

Comment:
@osi-scampbell Can you try with a distribution that sets 100% of the amount to another company. It seems that we are getting the wrong account: we get the "Due To/Due From" account instead of the expense account from the vendor bill line...

Reply:
@max3903 I can make it default to a different company. If we switch from that company back to our company do they get the correct account?